src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,147 @@
+package duke.parser;
+
+import duke.command.*;
+import duke.exception.DukeCommandException;
+import duke.exception.DukeException;
+import duke.exception.DukeInvalidArgumentException;
+import duke.exception.DukeMissingArgumentException;
+import duke.task.*;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeFormatterBuilder;
+import java.time.format.DateTimeParseException;
+import java.util.Locale;
+
+/**
+ * Parses input String into a more decipherable object.
+ */
+public class Parser {
+    /**
+     * Formats an input String into a LocalDateTime object.
+     */
+    public static final DateTimeFormatter formatter = new DateTimeFormatterBuilder()
+            .parseCaseInsensitive()
+            .appendPattern("yyyy-MM-dd HHmm")
+            .toFormatter(Locale.ENGLISH);
+
+    /**
+     * Parses a given inputCommand into a Command object for the ease of interpreting commands.
+     *
+     * @param inputCommand String String of input command.
+     * @return Command Returns any subclass of Command based on the input String.
+     * @throws DukeException
+     */
+    public static Command parse(String inputCommand) throws DukeException {
+        if (inputCommand.length() == 0) {
+            try {
+                throw new DukeCommandException("");
+            } catch (DukeCommandException e) {
+                System.out.println(e);
+                return new InvalidCommand();
+            }
+        }
+        String[] inputArray = inputCommand.split(" ");
+        String firstArg = inputArray[0];
+        if (inputArray.length == 1) {
+            if (Command.CommandType.BYE.equals(firstArg)) {
+                return new ByeCommand(inputArray);
+            } else if (Command.CommandType.LIST.equals(firstArg)) {
+                return new ListCommand(inputArray);
+            } else {
+                throw new DukeMissingArgumentException(firstArg);
+            }
+        } else {
```

Comment:
The condition under this loop seems to quite heavily nested. Do you think it would be better to abstract out the conditions to use a `switch` statement? 
Then maybe the logic relevant to the specialised checking for each type of task could also be moved to a separate class/done by the **Task** classes themselves?